src/Symfony/Component/Workflow/Metadata/MetadataBag.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Workflow\Metadata;
+
+/**
+ * MetadataBag is a container for key/value pairs.
+ *
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+final class MetadataBag
```

Comment:
I started with a simple array, then I thought I will be cleaner to have a class. And finally I added the final keyword in order to be able to update the code with ease (BC layer...)
Here is the rational ;)

I understand your point of view and I think I will remove this class ;)

Reply:
I mean it's good for typehinting for sure. My main annoyance is that user is stuck with very limited stock functionality with no way to extend it. If you decide to still have it instead of array, please at least mark it `@final` instead of making it `final`, so userspace can still extend it on its own risk and meanwhile report his use case for us to consider opening it up, instead of having his hands tied. This is standard way in symfony codebase. See https://github.com/symfony/symfony/issues/25788